yunwoo/baekjoon/수학/11050-이항-계수-1.js

```diff
@@ -0,0 +1,37 @@
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on("line", function (line) {
+  input.push(line);
+}).on("close", function () {
+  // 풀이
+  const getFactorial = (num) => {
+    let factorial = 1;
+
+    for (let i = 2; i <= num; i++) {
+      factorial *= i;
+    }
+
+    return factorial;
+  };
+
+  const [n, k] = input[0].split(" ").map(Number);
+  const nFactorial = getFactorial(n);
+  const kFactorial = getFactorial(k);
+  const denominatorFactorial = getFactorial(n - k);
+
+  console.log(nFactorial / (denominatorFactorial * kFactorial));
```

Comment:
```suggestion
  const getFactorial = (num) => {
    let factorial = 1;

    for (let i = 2; i <= num; i++) {
      factorial *= i;
    }

    return factorial;
  };

  const [n, k] = input[0].split(" ").map(Number);
  const nFactorial = getFactorial(n);
  const kFactorial = getFactorial(k);
  const denominatorFactorial = getFactorial(n - k);

  console.log(nFactorial / (denominatorFactorial * kFactorial));
```
- 풀이 잘봤습니다! 🙇‍♂️
- 팩토리얼을 함수로 빼서 공통으로 사용한 부분 너무 좋았습니다 :) 
- 아래와 같이 재귀함수를 이용한 팩토리얼 로직으로도 활용할 수 있습니다 :) 
- 변수명 고민을 하신 것 같은데 저는 단순히 molecule(분자), denominator(분모)로 이용했습니다.
```js
const factorial = (num) => {
  if (num < 2) return 1;
  return num * factorial(num - 1);
};

const [N, K] = input[0].split(" ").map(Number);

const molecule = factorial(N);
const denominator = factorial(K) * factorial(N - K);

console.log(molecule / denominator);
```

Reply:
- 재귀함수로도 가능하군요!! 👍
- 아하.. 두가지를 합쳐서 변수로 할당할 생각을 못했네요ㅎㅎ.. 코멘트 감사합니다🙇‍♂️